src/code/z_fbdemo_fade.c

```diff
@@ -1,17 +1,133 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_Start.s")
+#define THIS ((TransitionFade*)thisx)
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_Init.s")
+static Gfx sFadeSetupDL[] = {
+    gsDPPipeSync(),
+    gsSPClearGeometryMode(G_ZBUFFER | G_SHADE | G_CULL_BOTH | G_FOG | G_LIGHTING | G_TEXTURE_GEN |
+                          G_TEXTURE_GEN_LINEAR | G_LOD | G_SHADING_SMOOTH),
+    gsDPSetOtherMode(G_AD_NOTPATTERN | G_CD_BAYER | G_CK_NONE | G_TC_FILT | G_TF_BILERP | G_TT_NONE | G_TL_TILE |
+                         G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_1PRIMITIVE,
+                     G_AC_NONE | G_ZS_PIXEL | G_RM_CLD_SURF | G_RM_CLD_SURF2),
+    gsDPSetCombineLERP(0, 0, 0, PRIMITIVE, 0, 0, 0, PRIMITIVE, 0, 0, 0, PRIMITIVE, 0, 0, 0, PRIMITIVE),
+    gsSPEndDisplayList(),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_Destroy.s")
+TransitionInit TransitionFade_InitVars = {
+    TransitionFade_Init,   TransitionFade_Destroy, TransitionFade_Update,   TransitionFade_Draw,
+    TransitionFade_Start,  TransitionFade_SetType, TransitionFade_SetColor, NULL,
+    TransitionFade_IsDone,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_Update.s")
+void TransitionFade_Start(void* thisx) {
+    TransitionFade* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_Draw.s")
+    switch (this->type) {
+        case 0:
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_IsDone.s")
+        case 1:
+            this->timer = 0;
+            this->color.a = (this->direction != 0) ? 255 : 0;
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_SetColor.s")
+        case 2:
```

Comment:
are these `TransitionInstanceType`?

Reply:
Probably not unless it was the `TransitionInstanceType` - 1 since the enum starts at 1

Reply:
No they are not. Fade takes the `TransitionInstanceType` and converts it something that is purely internal (1 for regular transitions, 2 for flash, and 0 for nothing). I could add an enum, just wasn't sure if it is necessary.